python/tests/conftest.py

```diff
@@ -411,3 +560,80 @@ def get_df(ts, width, max_col_width):
         return pd.DataFrame(index=[pd.Timestamp(ts)], data={str(col): get_val(col) for col in cols})
 
     return get_df
+
+
+@pytest.fixture(scope="module")
+def azurite_port():
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    port = 10000
+    max_port = 65535
+    while port <= max_port:
+        try:
+            sock.bind(("", port))
+            sock.close()
+            return port
+        except OSError:
+            port += 1
+    raise IOError("no free ports")
+
+
+@pytest.fixture(scope="module")
+def spawn_azurite(azurite_port):
+    if sys.platform == "linux":
+        print("Spawning Azurite")
+        port = str(azurite_port)
+        temp_folder = tempfile.TemporaryDirectory().name
```

Comment:
I have tried it but sadly it is either session-scoped or function-scoped. Instead of having some workaround for tmpdir, I will keep it as it is.

Reply:
Thanks for the reminder.
Module is selected at the first place as each module could have its own Azurite so that a very basic "load-sharing" can be setup.
If it is a session-scope, the loading can easily make Azurite crash